lldb/bindings/interface/SBLockExtensions.i

```diff
@@ -0,0 +1,11 @@
+%extend lldb::SBLock {
+#ifdef SWIGPYTHON
+    %pythoncode %{
+        def __enter__(self):
+            return self
+
+        def __exit__(self, exc_type, exc_value, traceback):
+            self.Unlock()
```

Comment:
It's a bit unfortunate, that

```
lock = target.AcquireLock();

with lock:
   # we are correctly locked here

with lock:
   # we are no longer locked here :/
```

does not work as expected.

In Python, the convention seems to be that locks can be locked or unlocked, and entering a `with` scope` acquires the lock. We might want to model this after the Python's [Lock](https://docs.python.org/3/library/threading.html#lock-objects) or [RLock](std::unique_lock) objects, i.e. closer to a `std::unique_lock` than a `std::lock_guard`

WDYT?

Reply:
Can you think of a way to make this work in Python, without giving up on the RAII property on the C++ side? I my opinion the `SBLock` is a lot more important in C++ than it is in Python (where the GIL means you don't truly have multithreading anyway, though I know in the future there might be a mode without the GIL).

Reply:
What I can make work is this:
```
lock = target.AcquireLock();
[...]
lock.Unlock()
with lock:
   # we are correctly locked here
with lock:
   # we are correctly locked here
```

Reply:
Sounds good. If we would have an additional `GetLock` (beides `AcquireLock`) which returns the non-locked Lock, then I think we would have a full idiomatoc interface, also in Python